[18.0][MIG] quality_control_oca

AaronHForgeFlow opened the pull request:
Supersedes https://github.com/OCA/manufacture/pull/1374

it includes https://github.com/OCA/manufacture/pull/1491

cc @ForgeFlow

oihane committed:
Modulos de calidad sin migrar

pedrobaeza committed:
quality_control * PEP8 * Travis * Size in field (deprecated)

alfredoavanzosc committed:
no msg

pedrobaeza committed:
quality_control: Some improvement

oihane committed:
<quality_control> deleted constraint

pedrobaeza committed:
quality_control: Total refactorization with new API, README files, and new concepts. * Triggers for product category * ACLs * Tolerances in questions * one2many copyable * Fix some views * Error on same tolerance * Colors on inspection lines * Unify inspection header data assignment in set_test method * key name in set_test * Change icon set

oihane committed:
New module <quality_control_hr> closes #662 * Changes in access rules * Trigger line instead of test and user

pedrobaeza committed:
quality_control * Fix wizard set test

* Expand AUTHORS

* Fix permissions

ejans committed:
changed default references of fields.Datetime.now() to fields.Datetime.now

sysadminmatmoz committed:
Slovene translations added

oihane committed:
quality_control: * Changed permissions

* security modified
* some demo info added
* Added "partner_selectable" field, so that inspections can by made by partner
* Little view fixing, field was oe_edit_only instead of label
* Add partners to trigger lines
* Tests added
* Trigger lines with duplicated test cleaned

quality_control: Propose to OCA

quality_control: required changes

quality_control: Fixing coveralls

quality_control_stock: * added demo data * replaced unlink method by ondelete="cascade" * required changes

pedrobaeza committed:
quality_control_*: Make use of qty field

Making it a regular field, you can put any manual quantity in inspections

oihane committed:
quality_control: deleting qc.inspection does not delete its lines fixed

Commit:
[IMP] quality_control: decimal precision configurable + approving inspections only by quality control managers

LoisRForgeFlow committed:
[9.0][MIG] quality_control

Commit:
[FIX] quality_control: Several things:

* not take into account inactivated tests trigger lines
* show action_cancel button in inspections even if it is autogenerated

zamberjo committed:
[FIX] quality_control: Fix TypeError on new test category (#193)

`TestQualityControl` is failing on `QcTestTemplateCatergory._get_complete_name`
when `name` is set as `False`.

LoisRForgeFlow committed:
[9.0][FIX] quality_control: use odoo tag on views

SimoRubi committed:
[MIG] quality_control: Migration to 10.0

Commit:
[MIG] quality_control: Migration to 11.0

SalahAdDin committed:
[11.0][FIX] quality_control: Fixed dashboard icon. (#304)

LoisRForgeFlow committed:
[10.0][FIX] quality_control: move quality control configuration menu to quality_control module to avoid runbot warning

oihane committed:
[FIX] quality_control: created new notebook page to avoid activate variants + permissions (#325)

pedrobaeza committed:
[IMP] quality_control: App drawer icon

NachoAlesLopez committed:
[MIG] quality_control: Migration to 12.0

LoisRForgeFlow committed:
[12.0][FIX] quality_control: add user admin to quality control manager group by default.

Commit:
[UPD] Update module logo

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 98.1% (154 of 157 strings)

Translation: manufacture-12.0/manufacture-12.0-quality_control
Translate-URL: https://translation.odoo-community.org/projects/manufacture-12-0/manufacture-12-0-quality_control/pt/

NachoAlesLopez committed:
[FIX] quality_control

Fixed Quality Control Trigger 'trigger' selector in Product Template and Product Category.

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 77.1% (121 of 157 strings)

Translation: manufacture-12.0/manufacture-12.0-quality_control
Translate-URL: https://translation.odoo-community.org/projects/manufacture-12-0/manufacture-12-0-quality_control/hr/

ps-tubtim committed:
[IMP] quality_control: black, isort

ps-tubtim committed:
[MIG] quality_control: Migration to 13.0 + renaming to quality_control_oca

Co-authored-by: Carlos Roca <carlos.roca@tecnativa.com>
Co-authored-by: ps-tubtim <pimolnats@ecosoft.co.th>
Co-authored-by: Pedro M. Baeza < pedro.baeza@tecnativa.com>

oca-travis committed:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] quality_control_oca: Missing README piece

CarlosRoca13 committed:
[IMP] quality_control_oca: Added triggers on product.product views

oca-travis committed:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
quality_control_oca 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-13.0/manufacture-13.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-13-0/manufacture-13-0-quality_control_oca/

enriquemartin committed:
[14.0][MIG] quality_control_oca

oca-travis committed:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[UPD] README.rst

AaronHForgeFlow committed:
[FIX] quality_control_oca: inherit from mail tracking in qc_trigger_line

oca-travis committed:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
quality_control_oca 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-14.0/manufacture-14.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-14-0/manufacture-14-0-quality_control_oca/

cubells committed:
[IMP] quality_control_oca: add chatter to quality tests

oca-travis committed:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
quality_control_oca 14.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-14.0/manufacture-14.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-14-0/manufacture-14-0-quality_control_oca/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 77.3% (123 of 159 strings)

Translation: manufacture-14.0/manufacture-14.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-14-0/manufacture-14-0-quality_control_oca/it/

pedrobaeza committed:
[FIX] quality_control_oca: Protect product category view with group

For avoiding access errors when other users without quality quality
permissions wants to open the product category view.

OCA-git-bot committed:
quality_control_oca 14.0.1.4.0

mymage committed:
[IMP] quality_control_oca: black, isort, prettier

mymage committed:
[MIG] quality_control_oca: Migration to 15.0

Commit:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-15.0/manufacture-15.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-15-0/manufacture-15-0-quality_control_oca/it/

mymage committed:
[MIG] quality_control_oca: Migration to 16.0

Commit:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

angelmoya committed:
[MIG] quality_control_stock_oca: Migration to 16.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update quality_control_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/es/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

Commit:
[UPD] Update quality_control_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (162 of 162 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (162 of 162 strings)

Translation: manufacture-16.0/manufacture-16.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-quality_control_oca/it/

pedrobaeza committed:
[FIX] quality_control_oca: XML-ID must be fully qualified

Fixes #1167

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[MIG] quality_control_stock_oca: Migration to 16.0

TT47858

uolaizola committed:
[FIX] quality_control_oca add colspan attribute to show the inline tree view of the fields correctly

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] quality_control_oca: pre-commit auto fixes

Commit:
[MIG] quality_control_oca: Migration to 17.0

Commit:
[UPD] Update quality_control_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (162 of 162 strings)

Translation: manufacture-17.0/manufacture-17.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-17-0/manufacture-17-0-quality_control_oca/sv/

Commit:
[UPD] Update quality_control_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-17.0/manufacture-17.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-17-0/manufacture-17-0-quality_control_oca/

jans23 committed:
Translated using Weblate (German)

Currently translated at 74.3% (119 of 160 strings)

Translation: manufacture-17.0/manufacture-17.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-17-0/manufacture-17-0-quality_control_oca/de/

Commit:
[UPD] Update quality_control_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-17.0/manufacture-17.0-quality_control_oca
Translate-URL: https://translation.odoo-community.org/projects/manufacture-17-0/manufacture-17-0-quality_control_oca/

Trivedi-Vacha-SerpentCS committed:
[Mig][18.0] quality_control_oca: Migrated the module

LauraCForgeFlow committed:
[FIX] quality_control_oca: notes width in QC inspection form

diggy128 committed:
[16.0][IMP] quality_control_oca: Add kanban views for qc.inspection and qc.inspection.line

diggy128 committed:
[IMP] quality_control_oca: update methods signatures

Update create signature to use vals_list
Fix raise in unlink override. Replace with api.ondelete method
Allow superuser to override ondelete constraint for autogenerated inspection
Replace unnecessary elif

AaronHForgeFlow committed:
[MIG] quality_control_oca: Migration to v18

LoisRForgeFlow commented:
@AaronHForgeFlow I think this improvement https://github.com/OCA/manufacture/pull/1278 should be included during migration

LoisRForgeFlow commented:
@AaronHForgeFlow Another one to include here https://github.com/OCA/manufacture/pull/1508

LoisRForgeFlow commented:
~One more! https://github.com/OCA/manufacture/pull/1509~  EDIT: not yet, it has conflicts pending to fix before merging

AungKoKoLin1997 committed:
[IMP] quality_control_oca, quality_control_stock_oca
This commit adds date_done field, update views and filters and the timing field in the QC trigger line to enable
following scenarios:

- When timing is 'Before', an inspection is generated for each related
  move when a picking with the trigger is confirmed.
- When timing is 'Plan Ahead', a 'Plan' inspection is generated for
  each related move when a picking with the trigger is confirmed. A
  plan inspection is just a plan, and cannot be updated except for the
  date. A plan inspection gets converted into an executable inspection
  once the picking is done.

AaronHForgeFlow commented:
> @AaronHForgeFlow I think this improvement #1278 should be included during migration

> @AaronHForgeFlow Another one to include here #1508


Both added!

AndreuOForgeFlow commented on quality_control_oca/views/qc_inspection_view.xml:
```diff
@@ -0,0 +1,526 @@
+                                <div class="oe_kanban_bottom_right">
+                                    <div
+                                        class="me-2"
+                                        attrs="{'invisible': [('state', 'in', ['success', 'failed'])]}"
```
```suggestion
                                        invisible="state in ['success', 'failed']"
```

AaronHForgeFlow replied:
Fixed. Thank you.

LoisRForgeFlow commented:
> ~One more! #1509~ EDIT: not yet, it has conflicts pending to fix before merging

RE-EDIT :laughing: @AaronHForgeFlow this one has been fixed and merged in 16

LoisRForgeFlow commented:
@AaronHForgeFlow could you try to fix conflicts here?

AaronHForgeFlow commented:
@LoisRForgeFlow do you still see conflicts? In local I did not find any conflicts. In the Pull Request only PSC can see the conflicts :confused:

LoisRForgeFlow commented:
> @LoisRForgeFlow do you still see conflicts? In local I did not find any conflicts. In the Pull Request only PSC can see the conflicts 😕

I see this

![image](https://github.com/user-attachments/assets/d0adbfb9-78f8-4273-b2d9-4ea165d82c7b)

Not sure if this is going to be an issue merging with ocabot

AaronHForgeFlow commented:
The branch is already rebased, I am not sure what to do now :disappointed:

LoisRForgeFlow commented:
It doesn't show differences in my local... let's move forward with reviews and let's hope there is no problem while merging.

LoisRForgeFlow commented:
@AndreuOForgeFlow could you update your review?

ELNOGAL commented on quality_control_oca/views/qc_inspection_view.xml:
```diff
@@ -0,0 +1,526 @@
+    <record model="ir.actions.act_window" id="action_qc_inspection">
+        <field name="name">Inspections</field>
+        <field name="res_model">qc.inspection</field>
+        <field name="view_mode">tree,form,kanban</field>
```
```suggestion
        <field name="view_mode">list,form,kanban</field>
```

AaronHForgeFlow replied:
Fixed. Thanks.

oihane reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot migration quality_control_oca

AndreuOForgeFlow reviewed (approved):
LGTM!

LoisRForgeFlow reviewed (approved):
Thanks for attending all reviews :medal_military:

LoisRForgeFlow commented:
/ocabot merge nobump
